src/network/netdev/wireguard.c

```diff
@@ -527,6 +528,38 @@ int config_parse_wireguard_private_key(const char *unit,
 
 }
 
+int config_parse_wireguard_private_key_file(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        _cleanup_free_ char *key = NULL;
+        Wireguard *w;
+        int r;
+
+        assert(data);
+        w = WIREGUARD(data);
+        assert(w);
+
+        r = read_one_line_file(rvalue, &key);
```

Comment:
(hmm, whenever we do something similar in other code we have (for example EnvironmentFile= in unit files) we tend to read the file at the moment of use, not already at the moment we load the config. I wonder if that would be better here too? I don't have a clear opinion on that though...)